app/API/V1/Repositories/PasswordResetRepository.php

```diff
@@ -0,0 +1,136 @@
+<?php
+
+namespace App\API\V1\Repositories;
+
+use App\API\V1\Entities\PasswordReset;
+use App\API\V1\Entities\User;
+use App\Repositories\Repository;
+use App\Exceptions\PasswordResetException;
+use TempestTools\Scribe\Doctrine\Events\GenericEventArgs;
+
+/** @noinspection LongInheritanceChainInspection */
+class PasswordResetRepository extends Repository
+{
+    protected /** @noinspection ClassOverridesFieldOfSuperClassInspection */
+        $entity = PasswordReset::class;
+
+    /**
+     * @param GenericEventArgs $e
+     * @throws \App\Exceptions\PasswordResetException
+     */
+    public function preCreate(GenericEventArgs $e): void
+    {
+        $k = $e->getArgs();
+        $params = $k['params'];
+        /** @var PasswordReset $entity */
+        $roleRepo = $this->getEm()->getRepository(User::class);
+        $user = $roleRepo->find(array_keys($params['user']['read'])[0]);
```

Comment:
Note that this also will get refactored due to some changes I put on the ticket. Generally, it would be better to get the entity out of $k['entity']->getUser() (provided that is populated at this point).

Reply:
preCreate does not have $k['entity'] though.